experimental/dds/tree2/src/core/undo/undoRedoManager.ts

```diff
@@ -16,11 +17,13 @@ export class UndoRedoManager<TChange, TEditor extends ChangeFamilyEditor> {
 	 * @param repairDataStoryFactory - Factory for creating {@link RepairDataStore}s to create and store repair
 	 * data for {@link UndoableCommit}s.
 	 * @param changeFamily - {@link ChangeFamily} used for inverting changes.
+	 * @param getHead - Function for retrieving the head commit of the branch associated with this undo redo manager.
 	 * @param headUndoableCommit - Optional commit to set as the initial undoable commit.
 	 */
 	public constructor(
 		public readonly repairDataStoreProvider: IRepairDataStoreProvider,
 		private readonly changeFamily: ChangeFamily<TEditor, TChange>,
+		private readonly getHead: () => GraphCommit<TChange> | undefined,
```

Comment:
It should be the first, I think. A branch may not have any commits so we need to support not having a head available. I think it's fine to require a getHead function that can just always return undefined if there actually is no branch associated with the URM like for the mocks.

Reply:
That makes sense if branches could be in a headless state, but they never are. Even if a branch has no commits, it still has a head. Even the trunk, when empty, has a head. There is an `origin commit` (which the trunk branches off of) that is always present specifically for these kinds of edge cases. Having such an origin commit removes the "if the branch is empty, do this, otherwise, do this" sort of logic.

Reply:
Oh interesting. I see that you mentioned getLastCommit never returns undefined. I think that's what confused me in the first place!

Reply:
Oh yea, sorry... that function is left over from before we had a commit graph and we stored commits in arrays. I should have updated it.